packages/eslint-plugin/src/rules/no-floating-promises.ts

```diff
@@ -118,6 +125,10 @@ export default createRule<Options, MessageId>({
           expression = expression.expression;
         }
 
+        if (isKnownSafePromiseReturn(expression)) {
```

Comment:
Good call. Going with the same attitude we've had before: looking purely at the returned type, not what happened to be called.

Reply:
I'm confused because I still don't see a test case with a finally

Reply:
Oh, I see, you added a `.then(() => {})` which I agree with, and is a good test case to have. The `x.finally(() => {})` case is specifically in question, though, since it triggers the rule to recurse and check `x`.

Reply:
Right! Added it to ensure the rule still reports, as the `.finally()` gives back a `Promise` rather than the expected known safe type.